src/field/fieldperp.cxx

```diff
@@ -378,12 +338,11 @@ const FieldPerp sliceXZ(const Field3D& f, int y) {
   // Source field should be valid
   checkData(f);
 
-  FieldPerp result(f.getMesh());
+  FieldPerp result(f.getMesh(), f.getLocation(), y, f.getDirectionX(),
+                   f.getDirectionY(), f.getDirectionZ());
```

Comment:
Here we can't use `emptyFrom` because `f` is a `Field3D` while `result` is a `FieldPerp`. This only happens in a couple of places, so it didn't seem worth adding a template or set of functions to handle it.